src/override/malloc.cc

```diff
@@ -116,6 +116,36 @@ extern "C"
   }
 #endif
 
+#if !defined(__NetBSD__)
+  SNMALLOC_EXPORT int
+    SNMALLOC_NAME_MANGLE(reallocarr)(void* ptr, size_t nmemb, size_t size)
+  {
+    bool overflow = false;
+    auto& a = ThreadAlloc::get();
+    size_t sz = bits::umul(size, nmemb, overflow);
+    if (SNMALLOC_LIKELY(sz == 0))
```

Comment:
I am surprised this is the likely case.

Reply:
reallocarr cares about this case.

Reply:
By "cares", do you mean this is a common case, or just a supported case.  Adding `SNMALLOC_LIKELY` is making this do the compiler will optimise for `sz==0`, and make code slower if it isn't.

Reply:
sorry I misread you earlier indeed likely is unneeded here.